packages/headless/src/utils/highlight.ts

```diff
@@ -11,6 +11,8 @@ export interface HighlightKeyword {
   length: number;
 }
 
+type HighlightKeywordOrString = HighlightKeyword | string;
```

Comment:
I feel this change is trying to squeeze the string into the keyword path. Could we create a separate function called `highlightString` instead?

Reply:
The  idea behind that change is that I wanted to expose only one function ( `highlightString()` ) to the users.

The `highlights` property of the argument (`params.highlights`) has type `HighlightKeyword[]`. But, from a users perspective, the highlights received in a `Result` object (e.g. `Result.excerptHighlights`) is a `string[]`.

So, I wanted to export only **one** function that handles both types. 

I thought about having 2 interfaces that differs only on the `highlights` attribute type and create an overload function so we would still have "one" function exported, but I found that that you can't overload function in ts/js.

Reply:
Thinking about this, should we expose `HighlightParams.highlights` as `string[]` since thats what the API returns in the result? I think this would make sense. Then, we would cast it in `highlightString()` to have `HighlightKeywords[]`

Reply:
Hmm, where do we get `params.highlights` from? Are these on the result object as well? or are they just for query suggest?

On exposing a single function, I wouldn't just prioritize this, because there can be benefits to splitting things up. For example, it may make sense to have a `highlightResult` function which takes a result obj (instead of lower-level `string[]`) and returns a highlighted excerpt and title. We could have a separate `highlightQuerySuggestion` function,

Let's talk on a slack call and compare/contrast different approaches,